privilege/privileges/privileges.go

```diff
@@ -405,43 +405,50 @@ func (*UserPrivileges) CheckPasswordExpired(sessionVars *variable.SessionVars, r
 }
 
 // ConnectionVerification implements the Manager interface.
-func (p *UserPrivileges) ConnectionVerification(user *auth.UserIdentity, authUser, authHost string, authentication, salt []byte, sessionVars *variable.SessionVars) error {
+func (p *UserPrivileges) ConnectionVerification(user *auth.UserIdentity, authUser, authHost string, authentication, salt []byte, sessionVars *variable.SessionVars) (string, error) {
```

Comment:
This seems to close the door for binding internal task to resource group?
For example, DDL sessions would not be a login session, so what's the way to set resource group for them?

Reply:
Good point. The whole process is like this:
1. PD or stand-alone resource manager contains prebuilt resource groups like "default", "ddl", "analyze", etc
2. DBA creates new resource groups for normal user
3. and optionally binds user@host to some resource group

For normal front-end requests through login user, use the bound resource group or "default" resource group.
Internal operations like DDL, auto analyze, etc associate their resource group in session.sessionVars.ResourceGroupName and pass down to TiKV through grpc.

It's possible specified resource group doesn't exist, default resource group will be used.

Reply:
Why `ddl`, `analyze` need prebuilt resource groups? Does `ddl` from different users share the same resource group?

Reply:
IMO. We can manage internal operator in a prebuild groups first, and then make it can be customize later if needed.